test/RenPool.test.js

```diff
@@ -0,0 +1,338 @@
+
+      const amount = 1;
+      await renToken.connect(alice).approve(renPool.address, amount);
+      await expect(
+        renPool.connect(alice).deposit(amount)
+      ).to.be.revertedWith('RenPool: Amount surpasses bond');
+    });
+
+  });
+
+  describe('withdraw/fulfillWithdrawRequest', function () {
+
+    it('should withdraw properly', async function () {
+      const amount = 1;
+
+      const balance = await renToken.connect(alice).balanceOf(alice.address);
+
+      await renToken.connect(alice).approve(renPool.address, amount);
+      await renPool.connect(alice).deposit(amount);
+
+      await renPool.connect(alice).withdraw(amount);
+
+      expect(await renToken.connect(owner).balanceOf(renPool.address)).to.equal(0);
+      expect(await renToken.connect(owner).balanceOf(alice.address)).to.equal(balance);
+      expect(await renPool.balanceOf(alice.address)).to.equal(0);
+      expect(await renPool.totalPooled()).to.equal(0);
+    });
+
+    it('should withdraw after unlocking', async function () {
+      const balance = await renToken.connect(alice).balanceOf(alice.address);
+
+      await renToken.connect(alice).approve(renPool.address, POOL_BOND);
+      await renPool.connect(alice).deposit(POOL_BOND);
+      expect(await renPool.isLocked()).to.be.true;
+
+      await renPool.connect(nodeOperator).unlockPool();
+      expect(await renPool.isLocked()).to.be.false;
+
+      await renPool.connect(alice).withdraw(POOL_BOND);
+
+      expect(await renToken.connect(owner).balanceOf(renPool.address)).to.equal(0);
+      expect(await renToken.connect(owner).balanceOf(alice.address)).to.equal(balance);
+      expect(await renPool.balanceOf(alice.address)).to.equal(0);
+      expect(await renPool.totalPooled()).to.equal(0);
+    });
+
+    it('should fulfill withdraw properly', async function () {
+      const amount = 1;
+
+      const aliceBalance = await renToken.connect(alice).balanceOf(alice.address);
+      const bobBalance = await renToken.connect(bob).balanceOf(bob.address);
+
+      await renToken.connect(bob).approve(renPool.address, POOL_BOND);
+      await renPool.connect(bob).deposit(POOL_BOND);
+
+      expect(await renPool.isLocked()).to.be.true;
+
+      await renPool.connect(bob).requestWithdraw(amount);
+      expect(await renPool.withdrawRequests(bob.address)).to.equal(amount);
+
+      await renToken.connect(alice).approve(renPool.address, POOL_BOND);
+      await renPool.connect(alice).fulfillWithdrawRequest(bob.address);
+
+      expect(await renToken.connect(owner).balanceOf(renPool.address)).to.equal(POOL_BOND);
+      expect(await renToken.connect(alice).balanceOf(alice.address)).to.equal(aliceBalance.sub(amount));
+      // expect(await renToken.connect(bob).balanceOf(bob.address)).to.equal(bobBalance);
+      expect(await renPool.balanceOf(alice.address)).to.equal(amount);
+      // expect(await renPool.balanceOf(bob.address)).to.equal(0);
+      expect(await renPool.isLocked()).to.be.true;
```

Comment:
Good catch! Why bob's balance is not being updated properly?

Reply:
Ok, solidity code seems fine, there was some issues with the tests. I think it should the as follows:

```
      expect(await renToken.connect(bob).balanceOf(bob.address)).to.equal(bobBalance.sub(POOL_BOND).add(amount));
      expect(await renPool.balanceOf(bob.address)).to.equal(POOL_BOND.sub(amount));
```